crates/wash-lib/src/common.rs

```diff
@@ -1,64 +1,125 @@
+/// name
+///
+/// If the string is a valid actor ID, it will be returned unchanged. If it is not an ID, it will
+/// attempt to resolve an ID in the following order:
 ///
-/// If the string is a valid actor ID, it will be returned unchanged. Resolution works by checking
-/// if the actor_name or call_alias fields from the actor's claims contains the given string. If
-/// more than one matches, then an error will be returned indicating the options to choose from
+/// 1. The value matches the prefix of the ID of an actor
+/// 2. The value is contained in the call alias of an actor
+/// 3. The value is contained in the name field of an actor
+///
+/// If more than one matches, then an error will be returned indicating the options to choose from
 pub async fn find_actor_id(
     value: &str,
     ctl_client: &wasmcloud_control_interface::Client,
 ) -> Result<(ModuleId, Option<String>), FindIdError> {
-    if let Ok(id) = ModuleId::from_str(value) {
+    find_id_matches(value, ctl_client).await
+}
+
+/// Given a string, attempts to resolve an provider ID. Returning the provider ID and an optional
+/// friendly name
+///
+/// If the string is a valid provider ID, it will be returned unchanged. If it is not an ID, it will
+/// attempt to resolve an ID in the following order:
+///
+/// 1. The value matches the prefix of the ID of a provider
+/// 2. The value is contained in the name field of a provider
+///
+/// If more than one matches, then an error will be returned indicating the options to choose from
+pub async fn find_provider_id(
+    value: &str,
+    ctl_client: &wasmcloud_control_interface::Client,
+) -> Result<(ServiceId, Option<String>), FindIdError> {
+    find_id_matches(value, ctl_client).await
+}
+
+async fn find_id_matches<T: FromStr + ToString + Display>(
+    value: &str,
+    ctl_client: &wasmcloud_control_interface::Client,
+) -> Result<(T, Option<String>), FindIdError> {
+    if let Ok(id) = T::from_str(value) {
         return Ok((id, None));
     }
-
     // Case insensitive searching here to make things nicer
     let value = value.to_lowercase();
     // If it wasn't an ID, get the claims
     let claims = ctl_client
         .get_claims()
         .await
-        .map_err(|e| FindIdError::Error(anyhow::anyhow!("Unable to get claims: {}", e)))?;
+        .map_err(boxed_err_to_anyhow)
+        .context("Unable to get claims for lookup")?;
     let all_matches = claims
         .iter()
         .filter_map(|v| {
-            let id = v
+            let id_str = v
                 .get(CLAIMS_SUBJECT)
                 .map(|s| s.as_str())
                 .unwrap_or_default();
-            // If it isn't a module, just skip
-            let id = match ModuleId::from_str(id) {
+            // If it doesn't parse to our type, just skip
+            let id = match T::from_str(id_str) {
                 Ok(id) => id,
                 Err(_) => return None,
             };
```

Comment:
Do you think a warning might be reasonable here? Maybe not

Reply:
Probably not worth the warning because fetching claims returns _all_ claims. So we will always have some that fail to parse as this type